app/tests/unit_tests/test_settings.py

```diff
@@ -55,3 +57,24 @@ def test_error_on_duplicate_section_check(self):
         self.test_settings.change_configuration_path(test_whitelist_duplicate_section_file)
         result = settings.check_no_duplicate_key()
         self.assertIsInstance(result, DuplicateSectionError)
+
+    def test_error_when_config_file_does_not_exist(self):
+        settings.args.config = [test_config_that_does_not_exist]
+        with self.assertRaises(SystemExit) as cm:
+            settings.check_if_config_file_exists()
+        self.assertEqual(cm.exception.code, 2)
+
+    def test_error_when_config_file_is_a_directory(self):
+        settings.args.config = [test_config_that_is_a_directory]
+        with self.assertRaises(SystemExit) as cm:
+            settings.check_if_config_file_exists()
+        self.assertEqual(cm.exception.code, 2)
+
+    def test_error_when_config_file_exists(self):
+        settings.args.config = [test_config_that_exists]
+        raised = False
+        try:
+            settings.check_if_config_file_exists()
+        except:
```

Comment:
Don't use an except here without specifying which exception. This will even catch on a typo in your "check_if_config_file_exists" call which is not what you want :)